### PR TITLE
changed id from number to string

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* /bcgov/entity#5399

*Description of changes:*
- updated version to 1.3.2
- changed button id to index
- renamed enum Roles -> RoleTypes
- added "uuid" package for new ids
- changed person.officer.id to string

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: src/components/AddPeopleAndRoles/ListPeopleAndRoles.vue [hunk 1]
```diff
@@ -80,7 +80,7 @@
           <div class="actions">
             <span class="edit-action">
               <v-btn small text color="primary"
-                :id="'officer-' + officer.id + '-change-btn'"
```
Comment (severinbeauvais) [inline src/components/AddPeopleAndRoles/ListPeopleAndRoles.vue:83]: `officer.id` doesn't actually exist

File: src/components/AddPeopleAndRoles/ListPeopleAndRoles.vue [hunk 2]
```diff
@@ -80,7 +80,7 @@
...
+                :id="`officer-${index}-change-btn`"
```
Comment (severinbeauvais) [inline src/components/AddPeopleAndRoles/ListPeopleAndRoles.vue:83]: This will give
```
officer-0-change-btn
officer-1-change-btn
etc
```

**Review by cameron-freshworks — APPROVED** [review]
Your right, it was easy!

File: tests/unit/OrgPerson.spec.ts [hunk 3]
```diff
@@ -33,92 +33,92 @@ const cancelButtonSelector: string = '#btn-cancel'
...
-    'id': 0,
-    'firstName': 'Adam',
-    'lastName': 'Smith',
-    'middleName': 'D',
-    'orgName': '',
-    'partyType': 'Person',
-    'email': 'completing-party@example.com'
+  officer: {
+    id: '0',
+    firstName: 'Adam',
```
Comment (severinbeauvais) [inline tests/unit/OrgPerson.spec.ts:38]: BTW, this "object format" allows you to get auto-completion of the object's properties ;)

File: tests/unit/OrgPerson.spec.ts [hunk 4]
```diff
@@ -230,39 +228,38 @@ describe('Org Person component', () => {
...
-    expect((<HTMLInputElement>wrapper.find(orgNameSelector).element).value)
-      .toEqual(validOrgData['officer']['orgName'])
+    const wrapper: Wrapper<OrgPerson> = createComponent(validOrgData, -1, null)
+    expect((<HTMLInputElement>wrapper.find(orgNameSelector).element).value).toEqual(validOrgData.officer.orgName)
     await Vue.nextTick()
     expect(wrapper.find(doneButtonSelector).attributes('disabled')).toBeUndefined()
     wrapper.find(doneButtonSelector).trigger('click')
     await Vue.nextTick()
-    expect(getLastEvent(wrapper, addEditPersonEvent)).toStrictEqual(validOrgData)
+    expect(getLastEvent(wrapper, addEditPersonEvent).officer.orgName).toEqual(validOrgData.officer.orgName)
```
Comment (severinbeauvais) [inline tests/unit/OrgPerson.spec.ts:257]: I had to change this as the "new officer logic" assigned a random id!

Also on line 391 below.

File: tests/unit/PeopleAndRoles.spec.ts [hunk 5]
```diff
@@ -25,14 +25,11 @@ const btnStartAddCompletingParty: string = '#btn-start-add-cp'
...
 const btnAddCompletingParty: string = '#btn-add-cp'
 const btnAddCorp: string = '#btn-add-corp'
-const btnPanel: string = '.btn-panel'
 const appointForm: string = '.appoint-form'
 const checkCompletingParty: string = '.cp-valid'
 const checkDirector: string = '.dir-valid'
 const checkIncorporator: string = '.incorp-valid'
-const completingPartyRole = { 'roleType': 'Completing Party', 'appointmentDate': '2020-03-30' }
-const directorRole: string = 'Director'
-const incorporatorRole: string = 'Incorporator'
```
Comment (severinbeauvais) [inline tests/unit/PeopleAndRoles.spec.ts:35]: I deleted the unused properties.

**Pull request merged**